transforms/__testfixtures__/react-19-replace-default-props/props-not-destructured.output.tsx

```diff
@@ -0,0 +1,3 @@
+const C = (props) => {
+  return <>{props.text}</>;
+};
```

Comment:
This no longer sets `props.text` by default. The codemod should opt out here with a message (or insert a comment) that it couldn't handle this case.

Reply:
@amirabbas-gh Can we do something like this?

### Before:

```tsx
const C = (props) => {
  const someValue = props.text;
  const { text } = props;
  return <>{props.text}</>;
};
```

### After:

```tsx
const C = (props) => {
  const props_textWithDefault = typeof props.text === 'undefined' ? 'test' : props.text;
  const someValue = props_textWithDefault;
  const { text = props_textWithDefault  } = props;
  return <>{props_textWithDefault}</>;
};
```

Reply:
It won't be able to handle the case of computed member expressions, which is ok, and perhaps we should just leave a note/comment for such cases:

```tsx
const C = (props) => {
  const key = cond ? "text" : "title";
  
  return <>
    {/* Codemod: Please make sure this is not affected by removal of defaultProps */}
    {props[key]}
  </>;
};
```

Reply:
@eps1lon 
I have another proposal:
```suggestion
const C = (props) => {
  props = {...props, text: typeof props.text === 'undefined' ? 'default' : props.text};
  return <>{props.text}</>;
};
```

Reply:
This seems fine.

Reply:
@eps1lon 
I fixed this in:
[this commit](https://github.com/reactjs/react-codemod/pull/331/commits/121afe2d6b8ab3f886b4f354c37eb991b59c663f#diff-95da743b662aea1c7cf7c4f089d354e223b258496fa8dcfeced27d6cd2a2a89d)

Reply:
TypeScript is probably going to be the biggest blocker here. But codemodding that perfectly seems like a pretty huge task.